RecoTracker/MkFit/plugins/createPhase1TrackerGeometry.cc

```diff
@@ -0,0 +1,372 @@
+    ic.m_region_order[0] = TrackerInfo::Reg_Transition_Pos;
+    ic.m_region_order[1] = TrackerInfo::Reg_Transition_Neg;
+    ic.m_region_order[2] = TrackerInfo::Reg_Endcap_Pos;
+    ic.m_region_order[3] = TrackerInfo::Reg_Endcap_Neg;
+    ic.m_region_order[4] = TrackerInfo::Reg_Barrel;
+
+    {
+      SteeringParams &sp = ic.m_steering_params[TrackerInfo::Reg_Endcap_Neg];
+      sp.reserve_plan(3 + 3 + 6 + 18);
+      sp.fill_plan(0, 1, false, true);  // bk-fit only
+      sp.append_plan(2, true);          // pick-up only
+      sp.append_plan(45, false);
+      sp.append_plan(46, false);
+      sp.append_plan(47, false);
+      sp.fill_plan(48, 53);  // TID,  6 layers
+      sp.fill_plan(54, 71);  // TEC, 18 layers
+      sp.finalize_plan();
+    }
+    {
+      SteeringParams &sp = ic.m_steering_params[TrackerInfo::Reg_Transition_Neg];
+      sp.reserve_plan(3 + 4 + 6 + 6 + 8 + 18);
+      sp.fill_plan(0, 1, false, true);  // bk-fit only
+      sp.append_plan(2, true);
+      sp.append_plan(3, false);
+      sp.append_plan(45, false);
+      sp.append_plan(46, false);
+      sp.append_plan(47, false);
+      sp.fill_plan(4, 9);    // TIB,  6 layers
+      sp.fill_plan(48, 53);  // TID,  6 layers
+      sp.fill_plan(10, 17);  // TOB,  8 layers
+      sp.fill_plan(54, 71);  // TEC, 18 layers
+      sp.finalize_plan();
+    }
+    {
+      SteeringParams &sp = ic.m_steering_params[TrackerInfo::Reg_Barrel];
+      sp.reserve_plan(3 + 1 + 6 + 8);
+      sp.fill_plan(0, 1, false, true);  // bk-fit only
+      sp.append_plan(2, true);          // pickup-only
+      sp.append_plan(3, false);
+      sp.fill_plan(4, 9);    // TIB, 6 layers
+      sp.fill_plan(10, 17);  // TOB, 8 layers
+      sp.finalize_plan();
+    }
+    {
+      SteeringParams &sp = ic.m_steering_params[TrackerInfo::Reg_Transition_Pos];
+      sp.reserve_plan(3 + 4 + 6 + 6 + 8 + 18);
+      sp.fill_plan(0, 1, false, true);  // bk-fit only
+      sp.append_plan(2, true);          // pickup-only
+      sp.append_plan(3, false);
+      sp.append_plan(18, false);
+      sp.append_plan(19, false);
+      sp.append_plan(20, false);
+      sp.fill_plan(4, 9);    // TIB,  6 layers
+      sp.fill_plan(21, 26);  // TID,  6 layers
+      sp.fill_plan(10, 17);  // TOB,  8 layers
+      sp.fill_plan(27, 44);  // TEC, 18 layers
+      sp.finalize_plan();
+    }
+    {
+      SteeringParams &sp = ic.m_steering_params[TrackerInfo::Reg_Endcap_Pos];
+      sp.reserve_plan(3 + 3 + 6 + 18);
+      sp.fill_plan(0, 1, false, true);  // bk-fit only
+      sp.append_plan(2, true);          // pickup-only
+      sp.append_plan(18, false);
+      sp.append_plan(19, false);
+      sp.append_plan(20, false);
+      sp.fill_plan(21, 26);  // TID,  6 layers
+      sp.fill_plan(27, 44);  // TEC, 18 layers
+      sp.finalize_plan();
+    }
+  }
+
+  void setupIterationParams(IterationParams &ip, unsigned int it = 0) {
+    if (it == 0) {
+      ip.nlayers_per_seed = 4;
```

Comment:
why don't you make them configurable in python ?

Reply:
I think this is only used for internal validation code at the moment. The code here, in general, produces in-memry default representation that is then written out as JSON.

Reply:
This function is now removed as part of removing the hardcoded parameters that are also present in the JSON files.